full-service/src/db/txo.rs

```diff
@@ -1548,9 +1548,11 @@ impl TxoModel for Txo {
             .filter(transaction_logs::failed.eq(false))
             .filter(transaction_logs::finalized_block_index.is_not_null())
             .filter(transaction_logs::submitted_block_index.is_not_null())
-            .filter(not(transaction_logs::account_id
-                .nullable()
-                .eq(txos::account_id)));
+            .filter(
+                txos::account_id
+                    .is_null()
+                    .or(transaction_logs::account_id.nullable().ne(txos::account_id)),
+            );
```

Comment:
SQL  A != B does not match when one of A or B are NULL. An explicit check for a NULL txos.account_id was needed.

Reply:
I think this can be replaced with:
`.filter(transaction_logs::account_id.is_distinct_from(txos::account_id))`
See https://docs.diesel.rs/1.4.x/diesel/expression_methods/trait.PgExpressionMethods.html#method.is_distinct_from (`This behaves identically to the != operator, except that NULL is treated as a normal value.`)

Reply:
thanks for the pointer to this comparison operator! However, it seems to be postgres only or at least not available when the database layer is sqlite, as sqlite lacks the underlying SQL IS DISTINCT FROM.

Reply:
Oh you're right! I forgot full-service is sqlite 🤦

Reply:
yes AND diesel doesn't natively use it -- we could mix in some raw SQL but maybe better to just use the compound query.